WebAPI/Controllers/StudentController.cs

```diff
@@ -83,6 +85,101 @@ public async Task<IActionResult> GetStudentTimetable(string userEmail)
+            bool isValidGUID = Guid.TryParse(studentId, out Guid guid);
+            if (!isValidGUID)
+            {
+                return ErrorResponse($"Student id: {studentId} is not valid GUID.");
+            }
+
+            var student = await _studentService.FindByIdAsync(guid);
+            if (student == null)
+            {
+                return ErrorResponse($"Student with id: {studentId} does not exist.");
+            }
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {studentId} does not exist.");
+            }
+            DateTime dateNow = DateTime.Now;
+            StringBuilder sb = new StringBuilder();
+
+            sb.AppendLine("BEGIN:VCALENDAR");
+            sb.AppendLine("VERSION:2.0");
+            sb.AppendLine("CALSCALE:GREGORIAN");
+            sb.AppendLine("METHOD:PUBLISH");
+
+            sb.AppendLine("BEGIN:VTIMEZONE");
+            sb.AppendLine("TZID:Europe/Amsterdam");
+            sb.AppendLine("BEGIN:STANDARD");
+            sb.AppendLine("TZOFFSETTO:+0100");
+            sb.AppendLine("TZOFFSETFROM:+0100");
+            sb.AppendLine("END:STANDARD");
+            sb.AppendLine("END:VTIMEZONE");
+
+            var dayOfWeek = 0;
+            switch (dateNow.DayOfWeek)
+            {
+                case DayOfWeek.Sunday:
+                    dayOfWeek = 7;
+                    break;
+                case DayOfWeek.Monday:
+                    dayOfWeek = 1;
+                    break;
+                case DayOfWeek.Tuesday:
+                    dayOfWeek = 2;
+                    break;
+                case DayOfWeek.Wednesday:
+                    dayOfWeek = 3;
+                    break;
+                case DayOfWeek.Thursday:
+                    dayOfWeek = 4;
+                    break;
+                case DayOfWeek.Friday:
+                    dayOfWeek = 5;
+                    break;
+                case DayOfWeek.Saturday:
+                    dayOfWeek = 6;
+                    break;
+            }
+
+            foreach (var block in student.Timetable.AllBlocks)
+            {
+                Course course = await _courseService.FindByIdAsync(block.CourseId);
+
+                //add the event
+                sb.AppendLine("BEGIN:VEVENT");
+
+                dateNow = DateTime.Now;
+                dateNow = dateNow.AddDays(-dayOfWeek + block.Day.GetHashCode())
+                    .AddHours(-dateNow.Hour +block.StartHour)
+                    .AddMinutes(-dateNow.Minute);
+
+                //with time zone specified
+                sb.AppendLine("DTSTART;TZID=Europe/Amsterdam:" + dateNow.ToString("yyyyMMddTHHmm00"));
+                sb.AppendLine("DTEND;TZID=Europe/Amsterdam:" + dateNow.AddHours(block.Duration).ToString("yyyyMMddTHHmm00"));
+
+                // i did not find anything to retrieve start or end of semester so it will repeat 13 weeks from now
```

Comment:
We discuss it, if nothinh exist like this, add some constant to configuration file like TermStart

Reply:
i added summer and winter starts to appconfig and retrieve it from there